janusgraph-core/src/main/java/org/janusgraph/graphdb/transaction/StandardJanusGraphTx.java

```diff
@@ -1478,10 +1478,19 @@ public synchronized void rollback() {
     }
 
     private void releaseTransaction() {
-        //TODO: release non crucial data structures to preserve memory?
         isOpen = false;
         graph.closeTransaction(this);
         vertexCache.close();
+        indexCache.close();
+        addedRelations.clear();
```

Comment:
What is the complexity of 'clear' method?
If it is O(n) then I would rather don't call 'clear' at all because it may slow down transactions.
Do we reuse addedRelations after we clear it? If no, there is definitely no point for additional computation.
'addedRelations = null' will be much faster in such case and will release even more resources (no need to hold an empty  array in-memory).

Reply:
Yeah, calling close/clear does have some overhead, although relatively very small compared to other computations in the transaction. Simply assigning null sounds like a good plan! Let me open another PR to fix this.

Reply:
Created https://github.com/JanusGraph/janusgraph/pull/2457